usaspending_api/download/management/commands/generate_spark_download.py

```diff
@@ -0,0 +1,224 @@
+)
+from usaspending_api.download.lookups import JOB_STATUS_DICT, FILE_FORMATS, VALUE_MAPPINGS
+from usaspending_api.download.models import DownloadJob
+from usaspending_api.download.v2.request_validations import AccountDownloadValidator, DownloadValidatorBase
+
+DOWNLOAD_SPEC = {
+    "award_financial": {
+        "federal_account": {
+            "df_builder": AccountDownloadDataFrameBuilder,
+            "validator_type": AccountDownloadValidator,
+        }
+    }
+}
+
+
+class Command(BaseCommand):
+
+    help = "Generate a download zip file based on the provided type and level."
+
+    download_job_id: int
+    download_job: DownloadJob
+    download_level: str
+    download_query: str
+    download_source: DownloadSource
+    download_spec: Dict
+    download_type: str
+    download_validator_type: Type[DownloadValidatorBase]
+    file_format_spec: Dict
+    file_prefix: str
+    jdbc_properties: Dict
+    jdbc_url: str
+    logger: Logger
+    should_cleanup: bool
+    spark: SparkSession
+    working_dir_path: Path
+
+    def add_arguments(self, parser):
+        parser.add_argument("--download-type", type=str, required=True, choices=list(DOWNLOAD_SPEC))
+        parser.add_argument(
+            "--download-level",
+            type=str,
+            required=True,
+            choices=set(
+                download_level
+                for download_level_list in [DOWNLOAD_SPEC[key] for key in DOWNLOAD_SPEC]
+                for download_level in download_level_list
+            ),
+        )
+        parser.add_argument("--download-job-id", type=int, required=True)
+        parser.add_argument("--file-format", type=str, required=False, choices=list(FILE_FORMATS), default="csv")
+        parser.add_argument("--file-prefix", type=str, required=False, default="")
+        parser.add_argument("--skip-local-cleanup", action="store_true")
+
+    def handle(self, *args, **options):
+        extra_conf = {
+            # Config for Delta Lake tables and SQL. Need these to keep Dela table metadata in the metastore
+            "spark.sql.extensions": "io.delta.sql.DeltaSparkSessionExtension",
+            "spark.sql.catalog.spark_catalog": "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+            # See comment below about old date and time values cannot parsed without these
+            "spark.sql.legacy.parquet.datetimeRebaseModeInWrite": "LEGACY",  # for dates at/before 1900
+            "spark.sql.legacy.parquet.int96RebaseModeInWrite": "LEGACY",  # for timestamps at/before 1900
+            "spark.sql.jsonGenerator.ignoreNullFields": "false",  # keep nulls in our json
+        }
+
+        self.spark = get_active_spark_session()
+        spark_created_by_command = False
+        if not self.spark:
+            spark_created_by_command = True
+            self.spark = configure_spark_session(**extra_conf, spark_context=self.spark)
+
+        # Setup Logger
+        self.logger = get_jvm_logger(self.spark, __name__)
```

Comment:
We have moved away from using the jvm_logger and instead using the Python logger: `logger = logging.getLogger(__name__)`: https://github.com/fedspendingtransparency/usaspending-api/blob/2a6276d3d2dbd2e51648d9f745a545fb01cafbbc/usaspending_api/etl/management/commands/create_delta_table.py#L17

This is something we changed since doing this work.